### ISSUE TITLE
Android HueApp Geofence/ location shows aiohue_xxxx as user

### ISSUE THREAD

**Issue by angex** [home-assistant-libs/aiohue#160]
Hi

After installed this addon my Android phone has started to show aiohue_xxxx user in location using devices. This means that geofencing features (like turn lights on when reacing home) is not working while one device is always at home.
If I remove this 'aiohue_xxx' device from Android phone /Location based devices it will within minutes come back.

Is there somewhere a plugin setting wich related on this which can be turned of? 

BR
Janne

**Comment by balloob** [comment]
Aiohue is a library. It's used by an application that you're using which is not setting a correct user string.

**Comment by marcelveldt** [comment]
Actually, this is caused by the keep alive workaround but it shouldn't mess with the geofencing. At least it didn't so far.

**Comment by juliansteenbakker** [comment]
I can confirm that the aiohue user is blocking geofence from working.

I've tried to replicate this by turning off my Home Assistant instance, removing the aiohue user, and running a 'comming home' route via the route mocking application 'lockito' on android. I've tried this 5 times and all 5 times the 'comming home' scene was activated about 100 meters before my house.

After starting the Home Assistant instance it stopped working completely.

I understand that this may be a case of wrong implementation of the library by Home Assistant, but maybe someone else can take a deeper look in this.

**Comment by balloob** [comment]
I don’t think that we should re-open this issue, because it doesn’t feel like it adequately describes the issue. We could open a new one, if it is in the lib.

@marcelveldt is the workaround in Home Assistant or in the lib?

**Comment by juliansteenbakker** [comment]
I think the workaround is defined here: https://github.com/home-assistant-libs/aiohue/blob/58b24e2ad0dd212f34a780689d0d90aa050a80dd/aiohue/v2/controllers/events.py#L284

Of course i can open a new issue if that's desirable.

**Comment by marcelveldt** [comment]
Correct, that is the workaround. Problem is that the workaround is still needed. I'm kind of surprised you're the first one in all these months that has noticed this so I guess the geofence in Hue is not so popular within the HA users ;-)

I'll inform with Signify if there's any sight on the solution where there is a keep alive message sent by the EventStream.

**Comment by juliansteenbakker** [comment]
Yeah i'm pretty confused myself. I haven't really found any other threads about this issue except this [reddit post.](https://www.reddit.com/r/Hue/comments/x80pnx/whats_this_random_device_in_location_enabled/)

In the PR that introduced this error https://github.com/home-assistant-libs/aiohue/pull/93 you mentioned something about feature-toggling this function. Is it perhaps possible to introduce this so i can at least disable this keep-alive feature when i'm disabling the hue integration?

**Comment by marcelveldt** [comment]
Why not use geofencing from within HA instead of Hue ?
Since you're literally the only/first person mentioning this I'm not going to create a feature toggle, simply because that is a lot of work to implement into the entire path.

I'd rather fix the underlying issue, detection if the connection is still alive.
I'll have a look if we can adjust the workaround so the fake device is not at home.

**Comment by juliansteenbakker** [comment]
> Why not use geofencing from within HA instead of Hue ? Since you're literally the only/first person mentioning this I'm not going to create a feature toggle, simply because that is a lot of work to implement into the entire path.

I understand you'd want to fix the underlying issue. The reason I'm not using HA geofencing instead of hue is that i'm using more scene's of Hue that are not available in HA like mimicking presence, and i want to have one place to manage my automations for a certain product because otherwise i'll lose track of where all my automations are.

**Comment by vjpr** [comment]
> Since you're literally the only/first person mentioning this

I have this issue too. I have two users setup in Hue app to turn off all Hue lights. I was relying on this and didn't realize it wasn't working for a while.

**Comment by db77x** [comment]
Hi, I have the same problem with geofencing Hue App with aoihue_ . I think, Noth Pole as Location would help us.
It would be graet, if you can adjust the workaround.

**Comment by vjpr** [comment]
Disabling the Hue integration doesn't stop it doing this either.

**Comment by marcelveldt** [comment]
I will take a look at this today, I'll keep you informed.

@vjpr disabling the Hue integration will work if you also delete the aiohue user in the app

**Comment by angex** [comment]
Hi

Should this work now? I tried to re-install this and geofencing is still not working. I used Lockito to simulate going a way and arriving home and without hue integration geofening works ok. If I activate this (aiohue_ appears as a device on android app) geofencing stops wroking.

BR
Janne

**Comment by marcelveldt** [comment]
No, it is not released yet

**Comment by angex** [comment]
> No, it is not released yet

Any news when it will be released? I have had to uninstall this because I need phone's gefencing to work.

**Comment by obeynothing** [comment]
I think that many people don't even know that the problem is caused by home assistants, which is why there are so few reports. It also took me some time to find out.

Is there a solution in the meanwhile?

**Comment by marcelveldt** [comment]
The fix should be in HA release 2023.2 which beta is released today

**Comment by Mariusthvdb** [comment]
does that mean there is a geofence entity too? (in the old days, we had a device_tracker for the Hue app device)

**Comment by obeynothing** [comment]
> The fix should be in HA release 2023.2 which beta is released today

It has the same behavior

**Comment by nabeelr** [comment]
I'm also struggling with this issue.

I'm only noticing it now, because I had a roommate who never left the house for the last 9 months or so. They have recently moved out, and I am now noticing the issue.

For what it's worth, the only reason I think more people aren't noticing this issue is because Geofencing has been broken in iOS for the last few years meaning the Hue app on iOS doesn't reliably (if ever) trigger the lights as it should. Apple doesn't care to fix this issue, so most iOS users simply don't use the feature. I personally have since switched to Android because of this, and all has been well until this bug came up.

Considering Geofencing not working was a big enough bug that made me, a life-time iOS user, and once Apple employee change platforms, it should indicate how important it is for this to work properly.

This is a deal breaker if it's not fixed soon.

**Comment by nabeelr** [comment]
Aaannddd, now it's working. I don't know why. But it wasn't working for the last week and I tried a few different things. Now it seems to be working, and I'm not complaining, so long as it doesn't stop working again.

**Comment by nabeelr** [comment]
This issue has come back and gone away twice in the last 3 months, with HA 2026.4.x and 2026.6.x. Can this be handled in a way where a location device doesn't need to be added to our accounts?
